### ISSUE TITLE
Versioned sub-app API does not produce docs

### ISSUE THREAD

**Issue by Falinor** [feathersjs-ecosystem/feathers-swagger#67]
Hi,

Thanks for your great plugin!
Although it is great, I have been struggling to build docs for my API, built as a versioned sub-app.

```js
const feathers = require('feathers');
const rest = require('feathers-rest');
const memory = require('feathers-memory');
const bodyParser = require('body-parser');
const swagger = require('feathers-swagger');

const messageService = memory();

messageService.docs = {
  description: 'A service to send and receive messages',
  definitions: {
    messages: {
      "type": "object",
      "required": [
        "text"
      ],
      "properties": {
        "text": {
          "type": "string",
          "description": "The message text"
        },
        "useId": {
          "type": "string",
          "description": "The id of the user that send the message"
        }
      }
    }
  }
};

const api = feathers()
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({ extended: true }))
  .configure(rest())
  .use('/messages', messageService);

const app = feathers()
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({ extended: true }))
  .configure(rest())
  .configure(swagger({
    docsPath: '/docs',
    info: {
      title: 'A test',
      description: 'A description'
    },
    prefix: '/api/v1',
    versionPrefix: '/v1',
    uiIndex: true
  }))
  .use('/api/v1', api);

app.listen(3000);
```

It should serve the following URIs:
  - docs: `http://localhost:3000/docs`
  - API: `http://localhost:3000/api/v1/messages`

Is that supported and, if it is, how may I do that properly?

Thank you very much,

**Comment by christopherjbaker** [comment]
Your `.use(...)` should also have the prefix, like `.use('/api/v1/messages', messageService)`.

My swagger config is also using regular expressions, so that the interface can know the version number, rather than just the prefix. I do not know what happens if you stick to strings, however.
```javascirpt
prefix: /api\/v\d\//,
versionPrefix: /v\d/,
```

**Comment by Falinor** [comment]
If I do like that, I will have an URL looking like `/api/v1/api/v1/messages`. Or I am not understanding your point.
I do not want to expose messages through `/api/v1/messages`. I first want to mount `/api/v1` as a sub-app, and then, add `/messages` to the router.
It is a supported behavior by feathers but not feathers-swagger, is it?

**Comment by Falinor** [comment]
No offense but, any clue about that issue?

**Comment by christopherjbaker** [comment]
TLDR: You can create feathers apps this way, but services can only know about services on their same app, so feathers-swagger must be on the same app as your services.

feathers-swagger does not currently know about sub-apps. It is conceivable that it could be made to support that, though it would require changes to feathers as well, to detect when a feathers app is used, to maintain a list, and possibly store all the providers in the same place (or leave it up to each plugin to use the nested providers if desired).

If the above were to be made (would have to see if @daffl is in favor), your above code would still have to be modified. feathers-swagger must be added to the app before the services, so they use its provider. This would mean your `api` would have to be created after your `app`, but `app.use('/api/v1', api);` would have to happen before `api.use('/messages', messageService);`
